manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -113,29 +109,35 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
 
         newProcessor.setName(processorRequest.getName());
         newProcessor.setSubmittedAt(ZonedDateTime.now());
-        newProcessor.setStatus(BridgeStatus.ACCEPTED);
+        newProcessor.setStatus(ManagedResourceStatus.ACCEPTED);
         newProcessor.setBridge(bridge);
         newProcessor.setShardId(shardService.getAssignedShardId(newProcessor.getId()));
 
         ProcessorDefinition definition = new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, requestedAction, resolvedAction);
         newProcessor.setDefinition(definitionToJsonNode(definition));
 
-        Optional<ConnectorEntity> persist = createProcessorConnectorEntity(newProcessor, actionProvider, resolvedAction);
+        LOGGER.info("Processor with id '{}' for customer '{}' on bridge '{}' has been marked for creation",
+                newProcessor.getId(),
+                newProcessor.getBridge().getCustomerId(),
+                newProcessor.getBridge().getId());
+
+        createProcessorConnectorEntity(newProcessor, actionProvider, resolvedAction);
 
-        persist.ifPresent(c -> eventBus.requestAndForget(CONNECTOR_CREATED_EVENT, c));
+        workManager.schedule(newProcessor);
 
         return newProcessor;
     }
 
     @Transactional
-    public Optional<ConnectorEntity> createProcessorConnectorEntity(Processor newProcessor, ActionProvider actionProvider, BaseAction resolvedAction) {
+    // Connector should always be marked for creation in the same transaction as a Processor
+    public void createProcessorConnectorEntity(Processor newProcessor, ActionProvider actionProvider, BaseAction resolvedAction) {
         processorDAO.persist(newProcessor);
-        return connectorService.createConnectorEntity(resolvedAction, newProcessor, actionProvider);
+        connectorService.createConnectorEntity(resolvedAction, newProcessor, actionProvider);
```

Comment:
If no connectors are created then the applicable worker simply won't do anything.

Reply:
@lucamolteni Understood.. My comment was explaining the change in behaviour of `connectorService.createConnectorEntity(..)` that no longer returns anything (that used to be used to advise whether we needed to kick-off the provisioning of MC's).